youtube_dl/extractor/roosterteeth.py

```diff
@@ -86,20 +90,29 @@ def _real_extract(self, url):
         api_episode_url = self._EPISODE_BASE_URL + display_id
 
         try:
-            m3u8_url = self._download_json(
-                api_episode_url + '/videos', display_id,
-                'Downloading video JSON metadata')['data'][0]['attributes']['url']
+            video_json = self._download_json(
+                api_episode_url + '/videos', display_id)['data'][0]
+            m3u8_url = url_or_none(try_get(
+                video_json, [
+                    lambda j: j['attributes']['url'],
+                    lambda j: j['links']['master']],
+                compat_str))
         except ExtractorError as e:
             if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
                 if self._parse_json(e.cause.read().decode(), display_id).get('access') is False:
                     self.raise_login_required(
                         '%s is only available for FIRST members' % display_id)
             raise
 
+        if m3u8_url is None:
+            raise ExtractorError("Unable to find formats")
```

Comment:
There's a check in the caller for missing formats, so it's better to just return an empty format array.

Reply:
I've changed this to just return an empty format array and empty subtitles.